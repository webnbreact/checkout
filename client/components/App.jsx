import ReactDOM from 'react-dom';



export default App;