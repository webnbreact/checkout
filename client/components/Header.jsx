import React from 'react';

const Header = () => (
  <div>
    <div>
      <span>$XX</span>
      <span> per night</span>
    </div>
    <div>
      <button type="button">
        <span>
          <span>[stars images]</span>
          <span>[reviews count]</span>
        </span>
      </button>
    </div>
    <div>
      [divider line]
    </div>
  </div>
);

export default Header;
