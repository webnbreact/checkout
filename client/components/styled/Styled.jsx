import styled, { keyframes } from 'styled-components';

export const BookStyle = styled.div`
  margin-top: 15px;
  position: relative;
  z-index: 1;
`;

export const BoldText = styled.span`
  font-size: small;
  font-weight: 500;
`;

export const BoxIcon = styled.div`
  position: relative;
  float: right;
  right: 18px;
`;

export const BoxText = styled.div`
  font-size: medium;
  font-weight: 300;
  position: relative;
  float: left;
  left: 18px;
`;

export const BoxWrapper = styled.div`
  height: 42px;
  width: 324px;
  border: 1px solid #ebebeb;
  border-radius: 2px;
`;

export const Button = styled.button`
  border: none;
  cursor: pointer;
  padding: 0;
`;

export const ButtonQuest = styled(Button)`
  font-size: 15px;
  padding-left: 5px;
  &:hover {
    color: rgb(0, 132, 137);
  };
`;

export const ButtonSubmit = styled(Button)`
  width: 100%;
  height: 3em;
  background-color: #fd5a5f;
  font-size: 16px;
  font-weight: 500;
  color: #ffffff;
  border-radius: 4px;
  margin-top: 26px;
  margin-left: 1px;
`;

export const ButtonUnderline = styled(Button)`
  font-size: 13.5px;
  font-weight: 300;
  &:hover {
    text-decoration: underline;
  };
`;

export const FlagWidget = styled.div`
  text-align: center;
  margin: 15px;
`;

export const CenterText = styled(FlagWidget)`
  margin: 10px;
  font-weight: 500;
  font-size: 12px;
`;

export const CalSelect = styled.input`
  position: relative;
  display: inline-block;
  box-sizing: border-box;
  width: 148px;
  height: 100%;
  font-size: medium;
  padding-left: 18px;
  border: none;
`;

export const Divider = styled.hr`
  background-color: transparent;
  border: 0.5px solid #ebebeb;
`;

export const FlexBox = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: left;
`;

export const GuestsStyle = styled.div`
  margin-top: 12px;
`;

export const GuestDropdownStyle = styled.div`
  height: 329px;
  width: 100%;
  border: 0.5px solid #dedede;
  box-shadow: 0 0 1px black;
  overflow: hidden;
  position: relative;
  z-index: 10;
  background-color: white;
`;

export const GuestMenuBtn = styled(Button)`
  width: 100%;
  height: 100%;
  text-align: unset;
`;

export const Module = styled.div`
  box-sizing: border-box;
  min-height: 365px;
  border: 1px solid #ebebeb;
  margin-top: 66px;
  padding: 18px 26px;
  overflow: hidden;
`;

export const MainWrapper = styled.div`
  font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
  margin: 0 0 0 51px;
  overflow: hidden;
  color: #484848;
  width: 377px;
`;

export const Price = styled.span`
  font-size: x-large;
  font-weight: bold;
`;

export const PriceSummaryStyle = styled.div`
  box-sizing: border-box;
  height: 131px;
  width: 100%;
  margin-top: 20px;
  margin-bottom: 18px;
`;

export const PriceSummaryLine = styled.div`
  display: flex;
  justify-content: space-between;
  height: 24px;
  font-size: medium;
  font-weight: 300;
  padding-top: 3px;
  box-sizing: border-box;
`;

const slide = keyframes`
  0% {
    opacity: 0;
    transform: translateY(-150px);
  }
  50% {
    max-height: 150px;
    transform: translateY(0);
  }
  100% {
    opacity: 1;
    max-height: 150px;
    transform: translateY(0);
  }
`;

export const SlideDown = styled.div`
  animation: 2s ${slide};
  animation-delay: 2s;
  animation-fill-mode: forwards;
  overflow-y: hidden;
  max-height: 0;
  position: relative;
  z-index: 1;
`;

export const StarStyle = styled.span`
  color: #108489;
  position: relative;
  bottom: 2.5px;
`;

export const Text = styled.div`
  font-size: 14.75px;
  padding: 8px 0;
`;
