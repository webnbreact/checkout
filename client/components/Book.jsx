import React from 'react';
import Dates from './Dates.jsx';
import Guests from './Guests.jsx';
import PriceSummary from './PriceSummary.jsx';
import { ButtonSubmit, CenterText, BookStyle } from './styled/Styled.jsx';

class Book extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      checkedIn: false,    // default value should be false
      checkedOut: false,   // changed to true for testing
      checkinDate: null,
      checkoutDate: null,
      dropdown: 'none',     // -> can be 'guest', 'checkinCal', or 'checkoutCal'
      guests: {
        adults: 1,
        children: 0,
        infants: 0
      },
    };
    this.displayDropdown = this.displayDropdown.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  displayDropdown(item) {
    this.setState({
      dropdown: item,
    });
  }

  handleChange(e) {
    // TO BE FILLED OUT
  }

  render() {
    const { nightlyPrice, cleaningFee, serviceFee } = this.props.state;
    return (
      <BookStyle>
        <form action="">
          <div className="override-line-height">
            <Dates dropdown={this.state.dropdown} displayDropdown={this.displayDropdown} />
            <Guests dropdown={this.state.dropdown} displayDropdown={this.displayDropdown} handleChange={this.handleChange} />
          </div>
          <div>
            {this.state.checkedIn && this.state.checkedOut ? 
              <PriceSummary nightlyPrice={nightlyPrice} cleaningFee={cleaningFee} serviceFee={serviceFee} /> : null}
          </div>
          <div>
            <ButtonSubmit type="submit">
              <span>Request to Book</span>
            </ButtonSubmit>
          </div>
          <div>
            <CenterText>You won't be charged yet</CenterText>
          </div>
        </form>
      </BookStyle>
    );
  }
}

export default Book;
