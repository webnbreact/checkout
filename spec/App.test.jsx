import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../client/components/App.jsx';
import Header from '../client/components/Header.jsx';
import Book from '../client/components/Book.jsx';
import { Price, Button, Text } from '../client/components/styled/Styled.jsx';

const { headerData } = require('./sampleData.js');

describe('App component', () => {
  const wrapper = shallow(<App />, { disableLifecycleMethods: true });
  
  it('should render without crashing', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should store state for the room\'s price and fees', () => {
    const state = wrapper.state();
    expect(state.nightlyPrice).toBeDefined();
    expect(state.cleaningFee).toBeDefined();
    expect(state.serviceFee).toBeDefined();
  });

  it('should render the Header and Book components', () => {
    expect(wrapper.find(Header)).toHaveLength(1);
    expect(wrapper.find(Book)).toHaveLength(1);
  });
});

describe('Header component', () => {
  const wrapper = mount(<Header state={headerData} />);

  it('should contain a valid number for the nightly price', () => {
    const price = wrapper.find(Price).text();
    const priceDigits = price.slice(1);
    expect(Number(priceDigits)).not.toBe(NaN);
  });

  it('should contain a button that indicates the number of reviews', () => {
    const reviewsButton = wrapper.find(Button);
    const reviewCount = wrapper.find(Text).get(1).props.children;
    expect(reviewsButton).toHaveLength(1);
    expect(typeof reviewCount).toBe('number');
  });
});
