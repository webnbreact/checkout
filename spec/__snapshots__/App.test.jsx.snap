// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render without crashing 1`] = `
<div>
  <div>
    <Header />
    <Book />
    <div>
      <span>
        Report this listing
      </span>
    </div>
  </div>
</div>
`;
